### ISSUE TITLE
`direction=:x / :y` means the opposite in `hist` and `density`

### ISSUE THREAD

**Issue by aplavin** [JuliaPlots/Makie.jl#4113]
It's very confusing that `hist(;direction=:x)` corresponds to `density(;direction=:y)`:
![image](https://github.com/user-attachments/assets/3d6b23a8-aa8a-41e4-914c-9a74cc241cbf)
Probably, either of these should be flipped?

- [x] what version of Makie are you running? (`]st -m Makie`) latest
- [x] can you reproduce the bug with a fresh environment ? (`]activate --temp; add Makie`) yes
- [x] What platform + GPU are you on? any

**Comment by jkrumbiegel** [comment]
Hm yeah I agree that this is confusing. Ideally I would want all of these to have the same options but it's not clear what they should be. In this case the reasoning was that the "main" data  for hist is the height of the bars, but for density it's the positions of the samples on the x axis. So they are flipped in that sense. But I don't want to pretend that this is not confusing.

Ideally, whatever we refactored it to would be a rule that's simple and unambiguous to understand.

**Comment by jkrumbiegel** [comment]
What about `barplot(1:10, 1:10)`. And you didn't specify how you'd flip hist or density.

**Comment by jkrumbiegel** [comment]
How do you specify a horizontal barplot then, you can't differentiate via positional args.

**Comment by jkrumbiegel** [comment]
Yeah that's what I've been slowly tending towards as well, canonical version plus `flip` attribute. And the canonical version should have the x/y ordering or as close to it as possible.

Another option is to just have two recipes for these cases, which can internally be implemented with a switch. But then the horizontal barplot for example would have a different signature to stay with the x/y pattern. You'd give the horizontal extents first and then the categories/positions.
